interface/weno/transmitrx.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * File use to transmit Rx
+ *
+ * @package OpenEMR
+ * @link    http://www.open-emr.org
+ * @author  Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @copyright Copyright (c) 2019 Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @license https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+require_once "../globals.php";
+
+use OpenEMR\Rx\Weno\NewRx;
+
+if (isset($_GET['scripts'])) {
+    $list = filter_input(INPUT_GET, 'scripts', FILTER_VALIDATE_INT);
+} else {
+    echo xlt("Error: No prescriptions sent");
+    exit;
+}
+
+$meds = explode(",", $list);
+
+/**
+ * This look is to package the XML and bundle the prescriptions togther
+ */
+foreach ($meds as $med) {
+    $sendScript = new NewRx();
+
+    //Returns XML message to be transmitted to gateway
+    $payload = $sendScript->creatOrderXMLBody($med);
+    //These colons were added to have a way to separate the presciptions on the endpoint server
+    $payloads .= $payload.":::";
+}
+
+$request_url = 'https://apa.openmedpractice.com/apa/interface/weno/receivingrx_v2';
+$response = get_url($request_url, $payloads);
+
+//Transmit RX payload to gateway
+
+function get_url($request_url, $payloads)
+{
+    $headers = [
+        "Access-Control-Allow-Origin: *",
+        "Content-type: text/xml",
+    ];
+    file_put_contents("payload/".rand()."-payload.xml", $payloads);
+    $data = ['xml' => $payloads];
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, $request_url);
+    curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+    curl_setopt($ch, CURLOPT_POST, true );
+    curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
+    curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+    curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false);
+
+    $response = curl_exec($ch) or die(curl_error($ch));
```

Comment:
Rather than curl, recommend using the standard openemr wrapper for this kind of thing. See oeHttp class and example uses throughout codebase.

Reply:
Done

Reply:
Problem during testing. The XML response is not coming back through the oeHttp.

    $response = oeHttp::bodyFormat('body')
        ->usingHeaders($headers)
        ->post($request_url, $payloads);
    file_put_contents("payload/wenoresponse.txt", $response);
    return $response;
When I dump the response, it is empty.  

I did not find an example in the code where the oeHttp handle an XML response. The oeHttp is built to handle a JSON response only to my knowledge.

Reply:
um try $response->body()

Reply:
also add in some response status checks like:
if ($response->status() !== 200) {
            $errors .= "\nError:" . xlt("Http Error") . ": " . $response->getReasonPhrase() . " : " . $response->status();
        }

Reply:
$response->getReasonPhrase() returns OK
$response->status() returns 200

The message that is being returned looks like this.

<Message xmlns="https://wenoexchange.com/schema/POSTRX">
    <Header> 
        <SentTime>2020-02-11T06:02:03</SentTime>
    </Header>
    <Body><IFrameURL>https://test.wenoexchange.com/en/newrx/reviewrx?token=87d5bdbc58934df1bd91eef76cc11ab6</IFrameURL>
    </Body>
</Message>